doc/languages-frameworks/go.section.md

```diff
@@ -0,0 +1,156 @@
+# Go
+
+## Go modules
+
+The function `buildGoModule` builds Go programs managed with Go modules.
+It builds a [Go Modules](https://github.com/golang/go/wiki/Modules) through a two phase build:
+- An intermediate fetcher derivation. This derivation will be used to fetch all of the dependencies of the Go module.
+- A final derivation will use the output of the intermediate derivation to build the binaries and produce the final output.
+
+### Example for `buildGoModule`
+
+In the following is an example expression using `buildGoModule`, the following
+arguments are of special significance to the function:
+
+- `vendorSha256`: is the hash of the output of the intermediate fetcher
+   derivation. `vendorSha256` can also take `null` as an input. When `null` is
+   used as a value, rather than fetching the dependencies and vendoring them, we
+   use the vendoring included within the source repo. If you'd like to not have
+   to update this field on dependency changes, run `go mod vendor` in your
+   source repo and set `vendorSha256 = null;`
+- `runVend`: runs the vend command to generate the vendor directory. This is
+  useful if your code depends on c code and go mod tidy does not include the
+  needed sources to build.
+
+```nix
+pet = buildGoModule rec {
+  pname = "pet";
+  version = "0.3.4";
+
+  src = fetchFromGitHub {
+    owner = "knqyf263";
+    repo = "pet";
+    rev = "v${version}";
+    sha256 = "0m2fzpqxk7hrbxsgqplkg7h2p7gv6s1miymv3gvw0cz039skag0s";
+  };
+
+  vendorSha256 = "1879j77k96684wi554rkjxydrj8g3hpp0kvxz03sd8dmwr3lh83j"; # (<co xml:id='ex-buildGoModule-1' />
+
+  runVend = true; <co xml:id='ex-buildGoModule-2' />
```

Comment:
https://github.com/NixOS/nixpkgs/pull/105004#discussion_r530920788
Any ideas how we should do these callouts?

Reply:
There is no markdown equivalent.

Reply:
One of the proposed markdown implementations was [MDX](https://mdxjs.com/), which would allow us to use markdown while still using XML tags for things we want to express that are not expressible in markdown.

For now, we could keep the code listings containing callouts as DocBook elements (keeping the necessary evil localized) and resolve it later when a markdown implementation is chosen.

Reply:
I stripped them in this case. It does not add a lot of value here given the complexity it adds.